core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -129,12 +134,12 @@ public ProxyConfiguration(String name, int port, String userName, String passwor
         this(name,port,userName,password,noProxyHost,null);
     }
 
-    @DataBoundConstructor
     public ProxyConfiguration(String name, int port, String userName, String password, String noProxyHost, String testUrl) {
         this.name = Util.fixEmptyAndTrim(name);
         this.port = port;
         this.userName = Util.fixEmptyAndTrim(userName);
-        this.secretPassword = Secret.fromString(password);
+        String tempPassword = Util.fixEmptyAndTrim(password);
+        this.secretPassword = tempPassword != null ? Secret.fromString(tempPassword) : null;
         this.noProxyHost = Util.fixEmptyAndTrim(noProxyHost);
         this.testUrl = Util.fixEmptyAndTrim(testUrl);
         this.authenticator = newAuthenticator();
```

Comment:
Will never be an `Authenticator` with password set. Fix attempt in https://github.com/jenkinsci/jenkins/pull/4607/commits/81839233030ab7a0dbb0bc37a938a8c708f34771